Modules/Calendar/Calendar/DatePicker/View/DatePickerCollectionViewDataSource.swift

```diff
@@ -0,0 +1,130 @@
+import UIKit
+import RxDataSources
+import OtherServices
+import Utils
+import RxSwift
+import RxCocoa
+
+protocol DatePickerCollectionViewDataSourceDelegate: class {
+    func swipped(direction: SwipeDirection)
+}
+
+struct DatePickerItemsSingleSection: SectionModelType {
+
+    var identity: Int = 0
+    var items: [DatePickerDayViewModel]
+
+    init(items: [DatePickerDayViewModel]) {
+        self.items = items
+    }
+
+    init(original: DatePickerItemsSingleSection, items: [DatePickerDayViewModel]) {
+        self = original
+        self.items = items
+    }
+}
+
+// swiftlint:disable line_length
+class DatePickerCollectionViewDataSource: RxCollectionViewSectionedReloadDataSource<DatePickerItemsSingleSection> {
+
+    unowned var delegate: DatePickerCollectionViewDataSourceDelegate!
+
+    var currentPage: Int = 0
+    var collectionView: UICollectionView
+    var pageWidth: CGFloat {
+        collectionView.bounds.width
+    }
+    var pageCount: Int {
+        (sectionModels.first?.items.count ?? 0) / 7
+    }
+    var currentlySelectedDate: Date
+    var initialPageUpdate = true
+
+    init(collectionView: UICollectionView, currentlySelectedDate: Date) {
+        self.collectionView = collectionView
+        self.currentlySelectedDate = currentlySelectedDate
+
+        super.init(
+            configureCell: { _, collectionView, indexPath, calendarItem in
```

Comment:
Can we rename the `calendarItem` here, so it's clear we're talking about date picker stuff, and not `CalendarItem`?
```suggestion
            configureCell: { _, collectionView, indexPath, datePickerItem in
```